### ISSUE TITLE
DMI_LONG_BITS_TO_DOUBLE_INVOKED_ON_INT missed on literal values

### ISSUE THREAD

**Issue by csrma** [spotbugs/spotbugs#3531]
The SpotBugs detector `DMI_LONG_BITS_TO_DOUBLE_INVOKED_ON_INT` is intended to catch cases where an `int` value is (accidentally) passed to `Double.longBitsToDouble(long)`, which almost always produces nonsensical results. However, it only flags calls where the value is first stored in an `int` variable and loaded from the local variable.
```java
int value = 1234567890;
double d = Double.longBitsToDouble(value);
```
Literal usages such as `Double.longBitsToDouble(1234567890);` go undetected. From a users point of view, both are the same risk. Below is a minimal working example:
```java
public class LiteralTest {
    public static void main(String[] args) {
        double d = Double.longBitsToDouble(1234567890);
    }
}
```
**Expected outcome:** SpotBugs should report `DMI_LONG_BITS_TO_DOUBLE_INVOKED_ON_INT` on line 3.

Spotbugs: `4.9.3`
Java: `Jdk 17` and `Jdk 21`
Run Method: `CLI` as:
```bash
javac LiteralTest.java && \
    java -jar $SPOTBUGS_JAR -textui -effort:max *.class
```

**Comment by JuditKnoll** [comment]
I checked this one with openjdk jdk 21, and it's not reported because the compiler optimizes this one out, so the literal doesn't even appear in the bytecode as int, only as a long.
The first example with true positive hit is compiled to this bytecode:
```
       0: ldc           #7                  // int 1234567890
       2: istore_1
       3: iload_1
       4: i2l
       5: invokestatic  #8                  // Method java/lang/Double.longBitsToDouble:(J)D
       8: dstore_2
```

The second example without hit the relevant line is compiled to this bytecode:
```
       9: ldc2_w        #14                 // long 1234567890l
      12: invokestatic  #8                  // Method java/lang/Double.longBitsToDouble:(J)D
      15: dstore        4
```
SpotBugs works on the compiled bytecode and as such it can't recognize issues, which doesn't appear in the bytecode.
